From 3a9db1647f58babe861f442e074390d57a3e1081 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 5 Jan 2024 02:40:57 +0100
Subject: [PATCH] lib: Use integer port argument with mysql_real_connect
 (#3325)

The port argument is not a string, but a plain integer.  In the past, compilers ignore such type errors and produced a warning only, but this is changing, so this change also addresses a potential build failure.

Unlike  PQsetdbLogin() which uses char *, mysql_real_connect() uses unsigned int for the port.

Found as part of Porting To Modern C in the Fedora Project.
---
 db/drivers/mysql/db.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/db/drivers/mysql/db.c b/db/drivers/mysql/db.c
index 5e5748c11f5..16a7ec5600b 100644
--- a/db/drivers/mysql/db.c
+++ b/db/drivers/mysql/db.c
@@ -53,8 +53,9 @@ int db__driver_open_database(dbHandle *handle)
         db_get_login2("mysql", name, &user, &password, &host, &port);
 
         connection = mysql_init(NULL);
-        res = mysql_real_connect(connection, host, user, password,
-                                 connpar.dbname, port, NULL, 0);
+        res =
+            mysql_real_connect(connection, host, user, password, connpar.dbname,
+                               port != NULL ? atoi(port) : 0, NULL, 0);
 
         if (res == NULL) {
             db_d_append_error("%s\n%s", _("Connection failed."),
